src/main/java/eu/dissco/core/translator/terms/specimen/SpecimenName.java

```diff
@@ -24,7 +28,37 @@ public String retrieveFromDWCA(ArchiveFile archiveFile, Record rec) {
 
   @Override
   public String retrieveFromABCD(JsonNode unit) {
-    return super.searchAbcdForTerm(unit, abcdefgTerms);
+    var result = getSpecimenName(unit, abcdTerm);
+    if (result == null){
+      result = getSpecimenName(unit, abcdefgTerm);
+    }
+    return result;
+  }
+
+  private String getSpecimenName(JsonNode unit, Pair<String, String> term) {
+    var numberFound = 0;
+    while (true) {
+      if (unit.get(String.format(term.getLeft(), numberFound)) != null) {
+        var preffered = unit.get(String.format(term.getLeft(), numberFound)).asBoolean();
```

Comment:
`unit.get(String.format(term.getLeft(), numberFound))`: This looks like a pricey operation, and you do it twice in a row. Maybe store it as a variable and reference it? 

Also: "preferred" has 2 Rs and 1 F. (I mess this word up all the time)

Reply:
Good remark.
I like the format approach as it remains clear which term we used from abcd.
However, it seems that builder or concatenation is much faster.
https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important
Will check if/how to refactor this.